### PR TITLE
feat: added example to showcase OZ ERC1967Upgrade (#613)

### LINKED ISSUE #613: Add OZ erc1967upgrade example

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
- added VoteV1, VoteV2, ProxyVote solidity contracts
- added tests to cover the contracts

**Related issue(s)**:

Fixes #613 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/solidity/oz/ERC1967Upgrade/VoteProxy.sol [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    /**
+     * @dev Performs implementation upgrade with additional setup call if data is nonempty.
+     * This function is payable only if the setup call is performed, otherwise `msg.value` is rejected
+     * to avoid stuck value in the contract.
+     *
+     * Emits an {IERC1967-Upgraded} event.
+     */
+    function upgradeToAndCall(address newImplementation) external onlyProxyAdmin(_msgSender()){
+        ERC1967Utils.upgradeToAndCall(newImplementation, "");
```
Comment (Nana-EC) [inline contracts/solidity/oz/ERC1967Upgrade/VoteProxy.sol:66]: q: what's the empty string param?
Maybe add a comment saying what it is and why it's empty.

File: contracts/solidity/oz/ERC1967Upgrade/VoteV1.sol [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+     * @return true if the voter has already voted, false otherwise
+     */
+    function voted(address voter) public view returns (bool) {
+        return _voted[voter];
+    }
+
+    /**
+     * @dev Returns the version of the vote system
+     */
+    function version() external view returns(uint256) {
```
Comment (Nana-EC) [inline contracts/solidity/oz/ERC1967Upgrade/VoteV1.sol:75]: Add calls to this method in your tests to confirm v1 and v2

File: test/solidity/oz/ERC1967Upgrade/Vote.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    })
+
+    it('Should load VoteV1 into proxy address', () => {
+      expect(proxiedVoteV1.address).to.eq(voteProxy.address)
+    })
+
+    it('Should cast votes to the system', async () => {
+      await proxiedVoteV1.connect(voter).vote()
+
+      const voters = await proxiedVoteV1.voters()
```
Comment (Nana-EC) [inline test/solidity/oz/ERC1967Upgrade/Vote.js:173]: Maybe expand this test or add another that confirms more than 1 voter is returned and the list is populated as expected

**Review by Nana-EC — CHANGES_REQUESTED** [review]
LG
Some test suggestions

File: contracts/solidity/oz/ERC1967Upgrade/VoteProxy.sol (hunk 1 above)
Comment (quiet-node) [reply contracts/solidity/oz/ERC1967Upgrade/VoteProxy.sol]: Resolved. I have added a bytes argument to it to let users decide on the data content

File: contracts/solidity/oz/ERC1967Upgrade/VoteV1.sol (hunk 2 above)
Comment (quiet-node) [reply contracts/solidity/oz/ERC1967Upgrade/VoteV1.sol:83]: resolved.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**